src/DynamoCoreWpf/Views/Search/LibrarySearchView.xaml.cs

```diff
@@ -116,10 +116,13 @@ private void OnRequestCloseToolTip(object sender, EventArgs e)
         private void OnButtonMouseLeftButtonDown(object sender, MouseButtonEventArgs e)
         {
             var senderButton = e.OriginalSource as FrameworkElement;
-            var searchElementVM = senderButton.DataContext as NodeSearchElementViewModel;
+            if (senderButton != null)
+            {
+                var searchElementVM = senderButton.DataContext as NodeSearchElementViewModel;
 
-            if (searchElementVM != null)
-                dragDropHelper.HandleMouseDown(e.GetPosition(null), searchElementVM);
+                if (searchElementVM != null)
+                    dragDropHelper.HandleMouseDown(e.GetPosition(null), searchElementVM);
```

Comment:
Hi, @ramramps , I played for a while with your changes and I noticed, that main category doesn't collapse in search view. I.e. 
![image](https://cloud.githubusercontent.com/assets/8158404/9058067/226ce3d8-3aa7-11e5-8e2f-6d29ba06292f.png)
 try to click on "Geometry". It should collapse. 

I can propose add to `dragDropHelper` some clear method. Like:
```c#
internal void Clear()
        {
            startPosition = new Point();
            nodeViewModel = null;
        } 
```

And in `LibrarySearchView.xaml.cs` add it like:
```c#
if (searchElementVM != null)
      dragDropHelper.HandleMouseDown(e.GetPosition(null), searchElementVM);
else
      dragDropHelper.Clear();
```

Let me know what you think. Thanks!